frontend/src/pages/PipelineDetails.tsx

```diff
@@ -521,8 +546,14 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
 
     this.props.updateToolbar({ breadcrumbs, actions: toolbarActions, pageTitle });
 
+    const graph = await this._createGraph(templateString);
+    let reducedGraph = graph ? transitiveReduction(graph) : null;
+    if (graph && reducedGraph && compareGraphEdges(graph, reducedGraph)) {
+      reducedGraph = null; // disable reduction switch
```

Comment:
Personal opinion:
I think there's no need to disable the switch, when reduced graph is the same.
It's understandable -- when the graph is already simple, reducing it has no effect. We've got tooltip on the 
switch too to explain this.

What do you think? anyway, this is not a blocker

Reply:
I would leave it disabled so that it's obvious that this features is not available for that particular pipeline. Otherwise users might try clicking it anyway, see that nothing changes, and wonder what is happening. I wouldn't rely on people actually reading and understanding what's in the tooltip, we can just prevent any potential misunderstanding by disabling the switch